iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -1181,6 +1178,51 @@ public void preDropPartitions(org.apache.hadoop.hive.metastore.api.Table hmsTabl
     context.putToProperties(HiveMetaStoreClient.SKIP_DROP_PARTITION, "true");
   }
 
+  private static void validatePartitionSpec(SearchArgument sarg, Table icebergTbl,
+      Map<String, PartitionField> partitionFieldMap) {
+    for (PredicateLeaf leaf : sarg.getLeaves()) {
+      TransformSpec transformSpec = TransformSpec.fromStringWithColumnName(leaf.getColumnName());
+      Types.NestedField column = icebergTbl.schema().findField(transformSpec.getColumnName());
+      PartitionField partitionColumn =
+          partitionFieldMap.values().stream().filter(pf -> pf.sourceId() == column.fieldId()).findFirst().get();
+
+      if (!(partitionColumn.transform() == null && transformSpec.transformTypeString() == null) &&
+          !partitionColumn.transform().toString().equalsIgnoreCase(transformSpec.transformTypeString())) {
+        throw new UnsupportedOperationException(
+            "Invalid transform for column: " + transformSpec.getColumnName() +
+                " Expected: " + partitionColumn.transform() +
+                " Found: " + transformSpec.getTransformType());
+      }
+    }
+  }
+
+  private static UnboundPredicate<Object> getPartitionPredicate(Table icebergTbl, PartitionData partitionData,
+      PartitionField field, int index) {
+    String columName = icebergTbl.schema().findField(field.sourceId()).name();
+    TransformSpec transformSpec = TransformSpec.fromString(field.transform().toString(), columName);
+
+    switch (transformSpec.getTransformType()) {
```

Comment:
isn't that duplicate code in filter#getPartitionTransform?
can't we extract `Expressions.equal( <transform>, partitionData.get(index, Object.class))`

Reply:
done